docs/core/deploying/single-file.md

```diff
@@ -11,10 +11,40 @@ Bundling all application-dependent files into a single binary provides an applic
 
 Single File deployment is available for both the [framework-dependent deployment model](index.md#publish-framework-dependent) and [self-contained applications](index.md#publish-self-contained). The size of the single file in a self-contained application will be large since it will include the runtime and the framework libraries. The single file deployment option can be combined with [ReadyToRun](../tools/dotnet-publish.md) and [Trim (an experimental feature in .NET 5.0)](trim-self-contained.md) publish options.
 
-There are some caveats that you need to be aware for single-file use, first of which is the use of path information to locate a file relative to the location of your application. The <xref:System.Reflection.Assembly.Location?displayProperty=nameWithType> API will return an empty string, which is the default behavior for assemblies loaded from memory. The compiler will give a warning for this API during build time to alert the developer to the specific behavior. If the path to the application directory is needed, the <xref:System.AppContext.BaseDirectory?displayProperty=nameWithType> API will return the directory where the AppHost (the single-file bundle itself) resides. Managed C++ applications aren't well suited for single-file deployment and we recommend that you write applications in C# to be single-file compatible.
+## API incompatibility
+
+Some APIs are not compatible with single-file deployment and applications may require modification if they use these APIs. If you use a third-party framework or package, it's possible that they may also use one of these APIs and need modification. The most common cause of problems is dependence on file paths for files or DLLs shipped with the application.
+
+The table below has the relevant runtime library API details for single-file use.
+
+| API                            | Note                                                                   |
+|--------------------------------|------------------------------------------------------------------------|
+| `Assembly.Location`            | Returns an empty string.                                               |
+| `Module.FullyQualifiedName`    | Returns a string with the value of `<Unknown>` or throws an exception. |
```

Comment:
I did a bit of cleanup in this table. Do you have an idea as to which exception is thrown here? Saying "throws an exception" is pretty generic. 😄

Reply:
Yeah, I prefer we don't specify an exception type here so people can't take a dependency on it. The exact behavior of this API may change through runtime refactorings or modifications.